bentoml/_internal/frameworks/onnx.py

```diff
@@ -0,0 +1,403 @@
+from __future__ import annotations
+
+import os
+import typing as t
+import logging
+from typing import TYPE_CHECKING
+
+import attr
+import numpy as np
+
+import bentoml
+from bentoml import Tag
+from bentoml.models import ModelContext
+from bentoml.models import ModelOptions
+from bentoml.exceptions import NotFound
+from bentoml.exceptions import BentoMLException
+from bentoml.exceptions import MissingDependencyException
+
+from ..types import LazyType
+from ..utils.pkg import get_pkg_version
+from ..runner.utils import Params
+
+if TYPE_CHECKING:
+    from bentoml.types import ModelSignature
+    from bentoml.types import ModelSignatureDict
+
+
+try:
+    import onnx
+    import onnxruntime as ort
+except ImportError:  # pragma: no cover (trivial error checking)
+    raise MissingDependencyException(
+        """onnx is required in order to use module `bentoml.onnx`, install
+        onnx with `pip install onnx`. For more information, refer to
+        https://onnx.ai/get-started.html
+        `onnxruntime` or `onnxruntime-gpu` is also required by `bentoml.onnx`.
+        Refer to https://onnxruntime.ai/ for more information.
+        """
+    )
+
+
+MODULE_NAME = "bentoml.onnx"
+MODEL_FILENAME = "saved_model.onnx"
+API_VERSION = "v1"
+
+logger = logging.getLogger(__name__)
+
+if TYPE_CHECKING:
+    import torch
+
+    from .. import external_typing as ext
+    from ..external_typing import tensorflow as tf_ext
+
+    ProvidersType = list[str | t.Tuple[str, t.Dict[str, t.Any]]]
+    ONNXArgType = t.Union[
+        "ext.NpNDArray", "ext.PdDataFrame", "torch.Tensor", "tf_ext.Tensor"
+    ]
+
+
+try:
+    import importlib.metadata as importlib_metadata
+except ImportError:
+    import importlib_metadata
```

Comment:
```suggestion
```

Reply:
I think we still need `importlib_metadata.PackageNotFoundError`